File: 0330-patching-array/0330-patching-array.c
int minPatches(int* nums, int numsSize, int n) {
    int cost = 0;
    long upper = 0;
    int i = 0;

    while (upper < n) {
        if (i < numsSize && nums[i] <= upper + 1) {
            upper += nums[i];
            i++;
        } else {
            cost++;
            upper += upper + 1;
        }
    }

    return cost;
}

File: 0330-patching-array/0330-patching-array.py
class Solution:
    def minPatches(self, nums: List[int], n: int) -> int:
        cost,upper=0,0
        i=0
        while upper<n:
            if i<len(nums) and nums[i] <= upper+1:
                upper += nums[i]
                i +=1
            else:
                cost+= 1
                upper += upper+1

        return cost
